util/http/http_handler.cc

```diff
@@ -315,7 +315,7 @@ error_code HttpConnection::ParseFromBuffer(io::Bytes buf) {
   cntx.set_user_data(user_data_);
 
   while (!buf.empty()) {
-    ParserType parser{std::move(request)};
+    ParserType parser;
```

Comment:
Is `request` not used by the `parser`?

Reply:
parser returns an parsed requests few lines below. without this change, the parser has reused already parsed requests for parsing it again, so it's good I added multiple requests to the tests - they uncovered the bug.

Reply:
see line 325

Reply:
gotcha, nice!